### ISSUE TITLE
Access document inside autosuggest event scope

### ISSUE THREAD

**Issue by jasonshouse** [matteodem/meteor-easy-search#131]
Hello! I cannot figure out how to access the original Document data from my search results. I've tried passing in fields in "returnFields", but it doesn't seem to be returning anything.

HTML
{{> esAutosuggest index="geos" reactive="false" class="search-box" placeholder="City, Zip, etc." renderSuggestion=suggestion }}

JS
EasySearch.createSearchIndex('geos', {
  'use':'mongo-db',
  'field':['desc'],
  'collection': Geos,
  'limit': 7,
  'returnFields': ['feedId','geoId','type'],
  'sort': function(){
    return { 'type':-1, 'listingCount':-1}
  },
  'query': function(searchString,opts){
    var query = EasySearch.getSearcher(this.use).defaultQuery(this,searchString);
    return query;
  }
});

After a user selects an option from the drop down menu, I grab the value using: instance.get('autosuggestSelected'). However, only some of the Document is returned... it's "desc" field, "pos" and "id", but nothing else! Based on the 'returnFields' array above, shouldn't I be seeing those additional fields?

Here's my result object:
{
  id: "9stWdSikLu8ZSZ2wh",
  pos: 0,
  value: "Austin"   // <-- that is the 'desc' field
}

Thank you for all your help and contributions, your easy-search package has saved me SO much time!

**Comment by matteodem** [comment]
return fields is only for the original document data and does not have anything to do with the autosuggest itself. It decides which fields you can see as a user and which not. Can you rephrase your problem into one question please?

**Comment by jasonshouse** [comment]
Sure thing! I'm trying to access the original document data based off what the user selects from the autosuggest box. Depending on how the user interacts with the UI, the exact same search will return different data.

CLICK: If the user clicks their selection from the dropdown, I can access the original document data:
![image](https://cloud.githubusercontent.com/assets/2614763/5921266/59ebdfd2-a608-11e4-8084-5bbb95b8923c.png)

KEYPRESS: However, if the user picks their selection using the arrow keys and enter, I only get back a couple fields. This is the exact same search, the only difference is the keypress approach:
![image](https://cloud.githubusercontent.com/assets/2614763/5921289/82b6f1a4-a608-11e4-9eae-84a6a834f1ce.png)

How do I access the original document data via the keypress approach?

**Comment by matteodem** [comment]
You can call findOne on the collection itself. Are you working with events? What scope are you in when trying to access the docs?

**Comment by jasonshouse** [comment]
I am currently calling findOne on the collection, i just assumed there was a way to avoid that extra call since the Click approach doesn't require it.

I'm working within the form's submit event. I grab the EasySearch instance:
![image](https://cloud.githubusercontent.com/assets/2614763/5921646/04588f22-a60b-11e4-96cd-d13bad5e97d2.png)

and autoSuggest values:
![image](https://cloud.githubusercontent.com/assets/2614763/5921668/31cace8e-a60b-11e4-8f43-3ef0df68a8a2.png)

and then making the findOne call on the collection:
![image](https://cloud.githubusercontent.com/assets/2614763/5921705/74accc0c-a60b-11e4-9f3a-8fb5e9556205.png)

**Comment by matteodem** [comment]
Right now it is not possible. I hope that this temporary solution works for you now and I'll find a way to include the docs themselves inside the value.
